################################################################################
SHORT DESCRIPTION:
################################################################################
On system reset, exit qemu in the stubdom.

################################################################################
LONG DESCRIPTION:
################################################################################
The libxl toolstack uprev has cleaned up several parts of the stubdom, including
OpenXT's reliance on the qemu process in the stubdom to crash. In the domain
reboot case, qemu receives a reset signal and continues executing the main loop.
However, we end up destroying the stubdom anyway, which ultimately prevents
dmbus from properly closing the connections to surfman and inputserver,
resulting in an unreleased, dead domain.

This patch modifies the system reset handling code so that we simply exit the
main loop as if qemu received a shutdown request, allowing surfman and input
to disconnect their dmbus connections.

################################################################################
CHANGELOG
################################################################################
Created: Chris Rogers, rogersc@ainfosec.com, 18/10/2016

################################################################################
REMOVAL
################################################################################
This patch is required by libxl.

################################################################################
UPSTREAM PLAN
################################################################################
This is OpenXT specific. There is no plan to upstream this patch.

################################################################################
INTERNAL DEPENDENCIES
################################################################################

################################################################################
PATCHES
################################################################################
--- a/vl.c
+++ b/vl.c
@@ -1859,12 +1859,9 @@ static bool main_loop_should_exit(void)
     request = qemu_reset_requested();
     if (request) {
         pause_all_vcpus();
-        qemu_system_reset(request);
-        resume_all_vcpus();
-        if (!runstate_check(RUN_STATE_RUNNING) &&
-                !runstate_check(RUN_STATE_INMIGRATE)) {
-            runstate_set(RUN_STATE_PRELAUNCH);
-        }
+        /* Stubdom will be destroyed anyway, just exit the main loop cleanly */
+        qapi_event_send_shutdown(shutdown_caused_by_guest(request));
+        return true;
     }
     if (qemu_wakeup_requested()) {
         pause_all_vcpus();
